zenodo/modules/spam/tasks.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2020 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Forms for spam deletion module."""
+
+from __future__ import absolute_import, print_function
+
+from flask import current_app
+from celery import shared_task
+import json
+
+from zenodo.modules.spam import current_spam
+from invenio_records.models import RecordMetadata
+
+
+@shared_task(ignore_result=False)
+def check_metadata_for_spam(depid_value, dep_id):
+    if not current_app.config.get('ZENODO_SPAM_MODEL_LOCATION'):
+        return json.dumps(False)
+    deposit = RecordMetadata.query.get(dep_id)
+    is_spam = current_spam.model.predict(
```

Comment:
Basing the decision on the float values returned by the `model.predict_proba()` result might give us some better control over the spam threshold... We can then fine-tune the threshold via config if needed.